lib/charms/mongodb_libs/v0/mongodb_tls.py

```diff
@@ -223,11 +236,12 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
             sans=self._get_sans(),
         )
         logger.debug("Requesting a certificate renewal.")
+        self.charm.set_secret(scope, "csr", new_csr.decode("utf-8"))
+
         self.certs.request_certificate_renewal(
             old_certificate_signing_request=old_csr,
             new_certificate_signing_request=new_csr,
         )
-        self.charm.set_secret(scope, "csr", new_csr.decode("utf-8"))
```

Comment:
any reason why we set a secret before requesting a new cert?

Reply:
@delgod see PR description :) (Changes section 2)

Reply:
@delgod yes! see the PR description, Changes section 2. It also keeps consistent with the other ways in which we request new certs and saving CSRs

Reply:
> Small change made to lib/charms/mongodb_libs/v0/mongodb_tls.py in _on_certificate_expiring this change sets the csr before requesting a new one (like we do when we request new CSRs elsewhere in the code). this is important because if request_certificate_renewal triggers cert_available events on the other units before the secret is set, the other units will not have access to the CSR in the app data

`request_certificate_renewal` make a change to the TLS relation.
`set_secret` make a change to the peer relation.
Both changes become applied only when the event is fully finished.

Reply:
Smart, gotcha, thanks Mykola :)